### PR TITLE
Auto-sync vendor assets & Hugo module versions + copyedits to README and CONTRIBUTING

### LINKED ISSUE #1182: Ensure that versions of dependencies remain in sync
We need to ensure that versions of dependencies remain in sync across installation options, namely:

- Hugo modules
- NPM packages
- CDN

### LINKED ISSUE #1723: Automate the updating of vendor assets

### PR DESCRIPTION
**Pull request by chalin**
- Followup to #1217
- Fixes #1182
- Fixes #1723
- Creates a single `assets/_vendor` folder to hold all missing vendor assets (which for now only contains the bootstrap vendor file)
- Adds a script to sync the BS vendor file(s) from the BS NPM package, triggered by a post-install event
- Adds a script to sync Hugo modules based on the corresponding NPM package versions. Hence the NPM version are canonical
- Updates `CONTRIBUTING.md`:
  - Simplifies the steps in **Publishing a release** since some changes are not handled automatically. Also removed references to the `dependencies` module that was dropped via #1217
  - Other tweaks
- Explicit mention of code-of-conduct in README and `CONTRIBUTING.md`
- Closes #1724 by superseding it (I needed to commit to the docsy repo so that I could test from the PR branch)

### CODE REVIEW

**Comment by chalin** [comment]
@deining - I've tested this using `docsy-example`, fetching the commit from the branch of this PR, and it works AFAICT.

**Review by deining — CHANGES_REQUESTED** [review]
Spotted a typo which needs to be corrected.
LGTM otherwise, haven't tested it, though.

**Comment by chalin** [comment]
Oh dang, tripped on the buttons somehow (or was it GH Desktop?). Anyhow, I didn't mean to merge this yet. I'll fix the typo in a followup commit.

**Comment by chalin** [comment]
Yeah, it wasn't the GH interface that did the merge, because it didn't squash the commits, which I have enabled by default. It must have been the GH desktop. :-(

**Comment by chalin** [comment]
My typo fix even made it into main!

File: assets/_vendor/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+This `_vendor` folder exists to work around a known bug in Go’s module
+management. For details, see <https://github.com/golang/go/issues/37397>.
+
+DO NOT EDIT or manually override the files in this folder. They are
+automatically synchronized as installation time.
```
[the remark this answers is not in the document]
Comment (chalin) [reply assets/_vendor/README.md:5]: Yes, thanks! Fixed.

**Comment by chalin** [comment]
Ok, I've reverted the commit and will resubmit a PR.

**Pull request merged**